protobuf/options.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package solo.io.protocgenopenapi;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = "github.com/solo-io/protoc-gen-openapi/protobuf/options";
+
+// FieldOptions expose custom configuration at the field level.
+// The schema generator parses these options from the proto descriptors to generate appropriate OpenAPI schemas.
+message FieldOptions {
+    // Enable this option to treat this field as an unstructured object in the OpenAPI schema.
+    // This can be set to disable infinite recursion of proto fields.
+    optional bool type_object = 1;
+
+    // Enable this option to treat this field as any value in the OpenAPI schema.
+    // This can be set to disable infinite recursion when expanding references on recursive types.
+    // In addition, should be set when using `google.protobuf.Value`, which can accept any JSON value type:
+    // object, array, string, number, boolean, or null.
+    optional bool type_value = 2;
+}
+
+extend google.protobuf.FieldOptions {
+    // TODO: update extension number once it has been registered with the upstream Extension registry
+    optional FieldOptions options = 123456;
```

Comment:
even if we don't update it, should be choose something more random?

Reply:
I was planning on using this as a placeholder and filing a ticket with upstream proto extensions to reserve this, i.e., do it the formal way. 
I should create the upstream PR just so that this is not ambiguous.

Reply:
not needed anymore